static/js/util/profile_edit.js

```diff
@@ -49,36 +55,66 @@ export function boundSelectField(keySet, label, options) {
     errors,
     updateProfile,
   } = this.props;
-  let onChange = value => {
+  let onChange = option => {
     let clone = _.cloneDeep(profile);
-    _.set(clone, keySet, value? value.value : '');
+    let toStore;
+    if (option.value !== null && option.value !== undefined) {
+      toStore = option.value.props.value;
+    } else {
+      toStore = null;
+    }
+    _.set(clone, keySet, toStore);
     updateProfile(clone);
   };
-  return <div>
-    <Select
-      options={options}
-      value={_.get(profile, keySet)}
-      placeholder={label}
-      onChange={onChange} />
-    <span className="validation-error-text">{_.get(errors, keySet)}</span>
-  </div>;
+  let selectedValue = _.get(profile, keySet);
+  let selectedOption;
+  for (let option of options) {
+    if (option.value === selectedValue) {
+      selectedOption = option;
+      break;
+    }
+  }
+
+  let convertOption = option => {
+    if (option) {
+      return {
+        text: option.label,
+        value: <MenuItem primaryText={option.label} value={option.value}/>
+      };
+    } else {
+      return {};
+    }
+  };
```

Comment:
I feel like the current way is cleaner since the `<MenuItem ...` stuff is hidden away in this function, and we don't need to duplicate `option.label` like we do here.